shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -284,176 +330,124 @@ private void configureStatusBarForFullscreenFlutterExperience() {
-    if (flutterFragment == null) {
-      // No FlutterFragment exists yet. This must be the initial Activity creation. We will create
-      // and add a new FlutterFragment to this Activity.
-      flutterFragment = createFlutterFragment();
-      fragmentManager
-          .beginTransaction()
-          .add(FRAGMENT_CONTAINER_ID, flutterFragment, TAG_FLUTTER_FRAGMENT)
-          .commit();
-    }
+  @Override
+  protected void onResume() {
+    super.onResume();
+    lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+    delegate.onResume();
   }
 
-  /**
-   * Creates the instance of the {@link FlutterFragment} that this {@code FlutterActivity} displays.
-   * <p>
-   * Subclasses may override this method to return a specialization of {@link FlutterFragment}.
-   */
-  @NonNull
-  protected FlutterFragment createFlutterFragment() {
-    BackgroundMode backgroundMode = getBackgroundMode();
-
-    Log.d(TAG, "Creating FlutterFragment:\n"
-        + "Background transparency mode: " + backgroundMode + "\n"
-        + "Dart entrypoint: " + getDartEntrypoint() + "\n"
-        + "Initial route: " + getInitialRoute() + "\n"
-        + "App bundle path: " + getAppBundlePath() + "\n"
-        + "Will attach FlutterEngine to Activity: " + shouldAttachEngineToActivity());
-
-    return new FlutterFragment.Builder()
-        .dartEntrypoint(getDartEntrypoint())
-        .initialRoute(getInitialRoute())
-        .appBundlePath(getAppBundlePath())
-        .flutterShellArgs(FlutterShellArgs.fromIntent(getIntent()))
-        .renderMode(backgroundMode == BackgroundMode.opaque
-            ? FlutterView.RenderMode.surface
-            : FlutterView.RenderMode.texture)
-        .transparencyMode(backgroundMode == BackgroundMode.opaque
-            ? FlutterView.TransparencyMode.opaque
-            : FlutterView.TransparencyMode.transparent)
-        .shouldAttachEngineToActivity(shouldAttachEngineToActivity())
-        .build();
+  @Override
+  public void onPostResume() {
+    super.onPostResume();
+    delegate.onPostResume();
   }
 
-  /**
-   * Hook for subclasses to control whether or not the {@link FlutterFragment} within this
-   * {@code Activity} automatically attaches its {@link FlutterEngine} to this {@code Activity}.
-   * <p>
-   * For an explanation of why this control exists, see {@link FlutterFragment.Builder#shouldAttachEngineToActivity()}.
-   * <p>
-   * This property is controlled with a protected method instead of an {@code Intent} argument because
-   * the only situation where changing this value would help, is a situation in which
-   * {@code FlutterActivity} is being subclassed to utilize a custom and/or cached {@link FlutterEngine}.
-   * <p>
-   * Defaults to {@code true}.
-   */
-  protected boolean shouldAttachEngineToActivity() {
-    return true;
+  @Override
+  protected void onPause() {
+    delegate.onPause();
+    super.onPause();
+    lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE);
```

Comment:
Why is `super` called in between `delegate` and `lifecycle`? Might deserve a comment. Same with `onStop`

Reply:
Traditionally I invoke local behavior before super behavior in lifecycles after onResume(), but I figured that a literal lifecycle notification should take place after the super class updates itself. I can place the delegate call after the super call if you think that would be valuable for readability.

Reply:
I went ahead and re-ordered them to avoid confusion.